wear/src/main/java/com/example/k_dual/watchface/KDualCanvasRender.kt

```diff
@@ -211,30 +284,79 @@ class KDualCanvasRender (
+            CustomPaint.differenceTextPaint(unitF, robotoRegular, null)
+        }
         paint.getTextBounds(differenceText, 0, differenceText.length, textBounds)
         canvas.drawText(differenceText, rectLeft+rectWidth-boxPaddingHorizontal-textBounds.width(), rectCenterY-textBounds.exactCenterY(), paint)
     }
 
-    private fun drawBloodGlucose(canvas: Canvas, order: Number, value: Int) {
-        val paint = CustomPaint.bloodGlucoseTextPaint(unitF, robotoMedium)
+    private fun drawBloodGlucose(canvas: Canvas, order: Number?, value: Int) {
+        val paint = CustomPaint.bloodGlucoseTextPaint(unitF, isDualMode, robotoMedium)
         val valueText = value.toString()
         paint.getTextBounds(valueText, 0, valueText.length, textBounds)
 
-        val rectHalfHeight = unitF * 27f / 2f
-        val textX = width - mHorizontal - pHorizontal - unitF*69f/2
-        val textY = if (order == 1) { height/2 - gap - pBottom - rectHalfHeight + textBounds.height()/2 } else { height/2 + gap + pBottom + textBounds.height() - rectHalfHeight + textBounds.height()/2 }
+        val textX: Float
+        val textY: Float
+
+        if (isDualMode) {
+            val rectHalfHeight = unitF * 27f / 2f
+            textX = width - mHorizontal - pHorizontal - unitF*69f/2
+            textY = if (order == 1) {
+                        height/2 - gap - pBottom - rectHalfHeight + textBounds.height()/2
+                    } else {
+                        height/2 + gap + pBottom + textBounds.height() - rectHalfHeight + textBounds.height()/2
+                    }
+        } else {
+            textX = width/2
+            textY = height/2 + textBounds.height()/2
+        }
+
         canvas.drawText(valueText, textX, textY, paint)
     }
 
+    override fun onTapEvent(tapType: Int, tapEvent: TapEvent, complicationSlot: ComplicationSlot?) {
+        // For test blink effect
+        if (tapType == TapType.DOWN) {
+            blinkEffect(1)
+        }
+    }
+
+    // Function for low/high alert highlights
+    private fun blinkEffect(order: Number?) {
+        var blinkCount = 0
+        val blinkHandler = Handler(Looper.getMainLooper())
+
+        val runnable = object : Runnable {
+            override fun run() {
+                if (blinkCount == 3) {
+                    isUser1AlertOn = false
+                    isUser2AlertOn = false
+                    blinkHandler.removeCallbacks(this)
+                } else {
+                    val showMillis: Long = if (blinkCount % 2 == 0) { 400 } else { 100 }
+                    if (order == 2) { isUser2AlertOn = !isUser2AlertOn } else { isUser1AlertOn = !isUser1AlertOn }
+                    blinkHandler.postDelayed(this, showMillis)
+                    blinkCount += 1
+                }
+            }
+        }
+        blinkHandler.removeCallbacks(runnable)
+        blinkHandler.post(runnable)
+    }
+
+
 }
 class KDualAssets: Renderer.SharedAssets {
```

Comment:
[ChatGPT] **Bitmap Handling**: The bitmap for the arrow is decoded from resources every time the `KDualCanvasRender` is initialized. This can be inefficient if the watch face is re-created multiple times. You can improve this by using a lazy initialization within the `KDualAssets`:
```kotlin
class KDualAssets: Renderer.SharedAssets {
    val arrowUp by lazy {
        BitmapFactory.decodeResource(resources, R.drawable.arrow_up)
    }
    // ... other shared assets
}
```